app/javascript/crayons/AutocompleteTriggerTextArea/AutocompleteTriggerTextArea.jsx

```diff
@@ -0,0 +1,444 @@
+import { h } from 'preact';
+import { useRef, useLayoutEffect, useReducer, useEffect } from 'preact/hooks';
+import { forwardRef, createPortal } from 'preact/compat';
+import { UserListItemContent } from './UserListItemContent';
+import { useMediaQuery, BREAKPOINTS } from '@components/useMediaQuery';
+import { isInViewport } from '@utilities/viewport';
+
+import {
+  useTextAreaAutoResize,
+  getAutocompleteWordData,
+  getCursorXY,
+} from '@utilities/textAreaUtils';
+
+// Used to ensure dropdown appears just below search text
+const DROPDOWN_VERTICAL_OFFSET = '1.5rem';
+const EMPTY_STATE_MESSAGE = 'No results found';
```

Comment:
nit: It felt a bit off to see a flash of `no results found` before the actual results. I would prefer to see something along the line of `loading...` but I understand it will require a different implementation and it may not be a part of the scope.

Reply:
That's a really good point! I double-checked and it's not new to this refactor, so I'll create a separate issue for a later enhancement (maybe good be a good one for the community too!)